applications/ColossalEval/colossal_eval/dataset/safetybench_en.py

```diff
@@ -0,0 +1,137 @@
+
+LANGUAGE = 'English'  # 'Chinese'
+EVAL_NAME = 'safetybench'
+INST = lang2inst[LANGUAGE]
+INPUT_FORMAT = lang2input_format[LANGUAGE]
+FILES = lang2files[LANGUAGE]
+
+CHOICE_TEMP = ["A. {}", "B. {}", "C. {}", "D. {}"]
+IDX2CHOICE = {0: 'A', 1: 'B', 2: 'C', 3: 'D'}
+
+default_inference_kwargs = {
+    "calculate_loss": False,
+    "all_classes": ["A", "B", "C", "D"],
+    "language": LANGUAGE,
+    "pretrain": False,
+    "max_new_tokens": 32}
+
+def get_query_str(question, options, choices_templates=CHOICE_TEMP, pad=True):
+    # {'questions': 'what is xxx?\n', options: ['aaa', 'bbb', 'ccc', 'ddd'], ...}
+    # --> 'what is xxx?\nA. aaa\n'
+    query = ""
+    query += question if question.endswith("\n") else question + "\n"
+    num_choices = len(choices_templates)
+    choices = []
+    for idx, option in enumerate(options):
+        choices.append(choices_templates[idx].format(option + '\n'))  # e.g. "A. xxxx\n", "B. xxxx\n", ...
+    remain_choice = num_choices - len(choices)
+    if pad and remain_choice > 0:  # use NULL choice to pad choices to max choices number 
+        fake_choice = 'NULL'
+        for i in range(remain_choice, num_choices):
+            choices.append(choices_templates[i].format(fake_choice + '\n'))
+    query += ''.join(choices)
+    query = INPUT_FORMAT.format(query)
+    return query
+
+def process_test(sample_list):
+    test_dict = {}
+    for sample in sample_list:
+        category = sample['category']
+        if category not in test_dict:
+            test_dict[category] = {'data': [], 'inference_kwargs': default_inference_kwargs}
+        question = sample['question']
+        options = sample['options']
+        query_str = get_query_str(question, options, pad=True)
+        data_sample = {
+            'dataset': EVAL_NAME,
+            'split': 'test',
+            'category': category,
+            'instruction': INST,
+            'input': query_str,
+            'output': '',
+            'target': '',
+            'id': sample['id']
+        }
+        test_dict[category]['data'].append(data_sample)
+    return test_dict
+
+def process_dev(sample_dict):
+    dev_dict = {}
+    for category in sample_dict.keys():
+        dev_dict[category] = {'data': [], 'inference_kwargs': default_inference_kwargs}
+        sample_list = sample_dict[category]
+        for sample_id, sample in enumerate(sample_list):
+            idx = sample['answer']
+            question = sample['question']
+            options = sample['options']
+            query_str = get_query_str(question, options, pad=True)
+            data_sample = {
+                'dataset': EVAL_NAME,
+                'split': 'dev',
+                'category': category,
+                'instruction': INST,
+                'input': query_str,
+                'output': '',
+                'target': IDX2CHOICE[idx],
+                'id': sample_id
+            }
+            dev_dict[category]['data'].append(data_sample)
+    return dev_dict
```

Comment:
Can be merged into one function.

Reply:
Oh test and dev set are in different format, so I use two function to tackle them. The .json file for test set is a list of dict, and the file for dev set is a dict of list.